src/account/ModuleManagerInternals.sol

```diff
@@ -296,15 +307,15 @@ abstract contract ModuleManagerInternals is IModuleManager {
             for (uint256 i = 0; i < _validationData.preValidationHooks.length; ++i) {
                 bytes calldata hookData = hookUninstallDatas[hookIndex];
                 (address hookModule,) = ModuleEntityLib.unpack(_validationData.preValidationHooks[i]);
-                _onUninstall(hookModule, hookData);
+                onUninstallSuccess = _onUninstall(hookModule, hookData);
                 hookIndex++;
             }
 
             for (uint256 i = 0; i < _validationData.permissionHooks.length(); ++i) {
                 bytes calldata hookData = hookUninstallDatas[hookIndex];
                 (address hookModule,) =
                     ModuleEntityLib.unpack(toModuleEntity(_validationData.permissionHooks.at(i)));
-                _onUninstall(hookModule, hookData);
+                onUninstallSuccess = _onUninstall(hookModule, hookData);
```

Comment:
The output of these two calls will be ignored when the value of `onUninstallSuccess` is overwritten on line 341.

If we want the variable to mean "did any of the calls fail", we can change this to be `onUninstallSuccess = onUninstallSuccess && _onUninstall(...);`, so that any failure bubbles up as a false.

But maybe we don't want that, since it wouldn't reveal which inner call failed. We could choose to ignore inner onUninstall call failuress instead. Wdyt?

Reply:
Good catch.